web/pdf_viewer.js

```diff
@@ -259,6 +259,11 @@ class PDFViewer {
     this.linkService = options.linkService || new SimpleLinkService();
     this.downloadManager = options.downloadManager || null;
     this.findController = options.findController || null;
+
+    if (this.findController) {
+      this.findController.onIsPageVisible = pageNumber =>
+        this._getVisiblePages().ids.has(pageNumber);
```

Comment:
Unrelated to this patch, but could we just check the visibility of the given page without having the get all the visible pages ?

Reply:
I have no idea how feasible that would be, given the multitude of scroll- and spread-modes that the viewer supports.

However please note that this code does *not* run for most "findagain" event, thanks to [this pre-existing check](https://github.com/mozilla/pdf.js/pull/16479/files#diff-693ef40ef030840f29fc58d0ec7dc7111114f096a6962d131aac3df4186e4430R648).